swig/CMakeLists.txt

```diff
@@ -28,6 +28,20 @@ endif()
 include(GrSwig)
 include(GrPython)
 
+function(GR_UNIQUE_TARGET desc)
```

Comment:
What is the purpose of this redefinition of GR_UNIQUE_TARGET? Does it speed up SWIG part. It would be great as it takes currently more time than the rest of compilation.

Reply:
I should have documented this change better. I was trying to fix the parallel build on Fedora 26/27/28 using cmake 3.11.0/3.11.2 
```bash
# make -j 4
make[2]: *** No rule to make target 'swig/grgsm_swig.py', needed by 'swig/grgsm_swig.pyc'.  Stop.
make[1]: *** [CMakeFiles/Makefile2:749: swig/CMakeFiles/pygen_swig_72fe6.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  2%] Built target pygen_python_bd098
[  2%] Built target grgsm_swig_swig_doc
[  4%] Building CXX object lib/CMakeFiles/grgsm.dir/flow_control/uplink_downlink_splitter_impl.cc.o
[  4%] Building CXX object lib/CMakeFiles/grgsm.dir/misc_utils/burst_file_sink_impl.cc.o
........
[ 39%] Building CXX object lib/CMakeFiles/grgsm.dir/trx/trx_burst_if_impl.cc.o
[ 40%] Linking CXX shared library libgrgsm-0.41.4git.so
[ 69%] Built target grgsm
make: *** [Makefile:141: all] Error 2
```
The strange thing is that I'm not able to reproduce this failure on Debian Testing with cmake 3.11.2.
I'm going to debug this problem some more and either amend the commit or drop it from this PR.

Reply:
It turns out that in Debian the `gnuradio-dev` package distribute a modified version of the UseSWIG.cmake where the parallel build issue is fixed.  https://github.com/gnuradio/gnuradio/commit/99a09af05fda6d0bab0cf3724a1c6bf453c71bc7
```
$ dpkg -l gnuradio-dev
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                            Version              Architecture         Description
+++-===============================-====================-====================-====================================================================
ii  gnuradio-dev:amd64              3.7.12.0-3           amd64                GNU Software Defined Radio toolkit development
$ dpkg -L gnuradio-dev  | grep UseSWIG.cmake
/usr/lib/x86_64-linux-gnu/cmake/gnuradio/UseSWIG.cmake
```
But in Fedora/CentOS/RHEL the `gnuradio-devel` package does not distribute such macro, they delete all cmake macros while building the package
https://src.fedoraproject.org/rpms/gnuradio/blob/master/f/gnuradio.spec#_150-152

So I've decided to copy the UseSWIG.cmake from the gnuradio repository (commit 4433a7703fe3f5713c2200a0f7c11b13510f34cc) into the local repository as there are copies of other gnuradio's macros there already.